dnf/module/repo_module_dict.py

```diff
@@ -453,7 +453,9 @@ def get_modules_dir(self):
         modules_dir = os.path.join(self.base.conf.installroot,
                                    self.base.conf.modulesdir.lstrip("/"))
 
-        ensure_dir(modules_dir)
+        if self.base.conf.installroot != "/":
```

Comment:
I'd probably remove this condition and run ensure_dir() every time.
Consider a situation when dnf is not installed from RPM and someone runs it from git checkout.